### DISCUSSION TITLE
Integration broken - Please read for more info

### DISCUSSION THREAD

**Discussion by AfonsoFGarcia** [AfonsoFGarcia/hass-bluecon#6] [Announcements]
Right now, this integration is completely broken and will not work.

As of today, the library supporting this integration was refactored to not include any of the API keys, at the request of Fermax. This means that the integration itself is broken.

If you managed to install it before today, it will still run if you don't update it.

The integration will be updated to request the necessary information at setup but I will not be able to publish the API keys, so it will be necessary for anyone using it to obtain them through other means.

As Fermax has reached out, I will try to see with them if there's any other way of keeping this integration alive.

Until then, hold tight.

**Comment by AfonsoFGarcia** [comment]
Release 0.6.0 is now live. With this release, you can pass the necessary API information as part of the configuration, instead of being embedded into the code.

Only mandatory fields are the client ID and secret for the Fermax API. The remaining fields, related with the firebase cloud messaging API, are not mandatory, but without them there will be no notifications of when someone rings the door.

I will not be providing any documentation on how to get this info.

In the meanwhile, I've heard back from Fermax and they will try to provide some credentials for us, but no promises made.

**Comment by jgpazos70** [comment]
> Right now, this integration is completely broken and will not work.
> 
> As of today, the library supporting this integration was refactored to not include any of the API keys, at the request of Fermax. This means that the integration itself is broken.
> 
> If you managed to install it before today, it will still run if you don't update it.
> 
> The integration will be updated to request the necessary information at setup but I will not be able to publish the API keys, so it will be necessary for anyone using it to obtain them through other means.
> 
> As Fermax has reached out, I will try to see with them if there's any other way of keeping this integration alive.
> 
> Until then, hold tight.

Hello, thank you very much for this integration. I upgraded before seeing the announcement and I lost the integration this morning.

How can we get those API keys or any other information to configure the integration? I can not see any option inside Fermax Blue APP to get them... Are them specific for each user or global to every users?

If I try to install old versions I can not get them to work again, is there any way to get an old version working again?

Regards.

**Reply by AfonsoFGarcia** [reply]
Sorry, there is no official way of getting any API keys and from my last contact with Fermax it's very very far away in their roadmap to do it (if they'll do it at all) to have an open API.

At their request, I removed all the API keys from the code of the underlying library. To me it feels against it to then explain how to obtain them so I will not be doing it.

I'm still hoping to find a way with them to keep this alive but until then it will be broken.

Regarding getting the old version working, it is not possible as all the versions of the underlying library containing the API keys were unpublished as well.

**Reply by jichizhang** [reply]
I won't link it here but you can find the API keys somewhere on the internet.

**Reply by wihlke** [reply]
@AfonsoFGarcia  No updates from Fermax I suppose?

**Comment by viseniv** [comment]
for me it never worked. 
never creates the config files.

**Comment by viseniv** [comment]
I found a python script that i can at least open the doors. Onky Fermax Blue user and password neded.

**Reply by c1328** [reply]
Interesting! Could you give us more details? Maybe a link to that script?

**Reply by viseniv** [reply]
https://github.com/marcosav/fermax-blue-intercom

**Reply by c1328** [reply]
Thanks! But what is the trick to get a Wi-Box running with Fermax Blue?

**Comment by Lagaf2024** [comment]
Any update ? don´t work ?

**Comment by Mariocoleguilla** [comment]
Any news on this? I currently have a Fermax VEO Duox Plus (9445) and I'm thinking on getting the same one but with Wi-Fi (a little too expensive though) to get access to the Fermax Blue cloud and integrate it with Home Assistant since I don't see any other way to know (via HA) when someone rings. Thank you!

**Comment by patrikulus** [comment]
Hi guys, with some help of AI I have moved the logic from the script linked by @viseniv to this extension. Tested on my VEO XS and it works fine. You can check my fork here: https://github.com/patrikulus/hass-bluecon
